worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionFilesSorter.java

```diff
@@ -290,6 +285,11 @@ public FileInfo getSortedFileInfo(
         }
       }
 
+      if (sortExceptions.containsKey(shuffleKey)
+          && sortExceptions.get(shuffleKey).containsKey(fileId)) {
+        throw sortExceptions.get(shuffleKey).get(fileId);
```

Comment:
Can there be a concurrent cleanup (which can result in NPE) here ?
If yes, we should try to make this atomic, something like:
```
  AtomicReference<IOException> sortExceptionRef = new AtomicReference<>(null);
  sortExceptions.compute(shuffleKey, current -> {
    if (null != current) {
      sortExceptionRef.set(current.get(fileId);
    }
    return current;
  });
  if (null != sortExceptionRef.get()) throw sortExceptionRef.get();
```